packages/flame/lib/src/components/input/joystick_component.dart

```diff
@@ -70,29 +72,37 @@ class JoystickComponent extends HudMarginComponent with Draggable {
   }
 
   @override
-  Future<void> onLoad() async {
-    await super.onLoad();
-    knob.anchor = Anchor.center;
-    knob.position.add(size / 2);
-    _baseKnobPosition = knob.position.clone();
-    if (background != null) {
-      add(background!);
-    }
-    add(knob);
+  @mustCallSuper
+  void onMount() {
+    assert(
+      background != null,
```

Comment:
Background can be null, it is only the knob that is mandatory.

Reply:
Done.